R/matching.R

```diff
@@ -88,6 +95,30 @@ estimate_weights <- function(data, centered_colnames = NULL, start_val = 0, meth
   wt <- exp(EM %*% alpha)
   wt_rs <- (wt / sum(wt)) * nrow(EM)
 
+  # bootstrapping
+  if (!is.null(nr.boot.iteration)) {
+    set.seed(set.seed.boot)
+    rowid_in_data <- which(!ind)
+    outboot <- lapply(1:nr.boot.iteration, function(k) {
+      boot_rows <- sample(x = 1:nrow(EM), size = nrow(EM), replace = TRUE)
+      boot_rowid <- rowid_in_data[boot_rows]
+      boot_EM <- EM[boot_rows, , drop = FALSE]
+      boot_opt <- optim(
+        par = alpha,
+        fn = objfn, gr = gradfn,
+        X = boot_EM,
+        method = method,
+        control = list(maxit = 300, trace = 0), ...
```

Comment:
I guess the dots would be most useful for the control parameters. Maybe we can improve this with the non-bootstrap optim call as well.
```suggestion
        control = list(maxit = 300, trace = 0, ...),
```